core/types/transaction_marshalling.go

```diff
@@ -94,6 +104,35 @@ func (t *Transaction) MarshalJSON() ([]byte, error) {
 		enc.V = (*hexutil.Big)(tx.V)
 		enc.R = (*hexutil.Big)(tx.R)
 		enc.S = (*hexutil.Big)(tx.S)
+	case *ShutterTx:
+		enc.ChainID = (*hexutil.Big)(tx.ChainID)
+		enc.Nonce = (*hexutil.Uint64)(&tx.Nonce)
+		enc.Gas = (*hexutil.Uint64)(&tx.Gas)
+		enc.MaxFeePerGas = (*hexutil.Big)(tx.GasFeeCap)
+		enc.MaxPriorityFeePerGas = (*hexutil.Big)(tx.GasTipCap)
+		enc.EncryptedPayload = (*hexutil.Bytes)(&tx.EncryptedPayload)
+		enc.L1BlockNumber = (*hexutil.Uint64)(&tx.L1BlockNumber)
+		enc.BatchIndex = (*hexutil.Uint64)(&tx.BatchIndex)
+		enc.To = t.To()
+		enc.V = (*hexutil.Big)(tx.V)
+		enc.R = (*hexutil.Big)(tx.R)
+		enc.S = (*hexutil.Big)(tx.S)
+	case *BatchTx:
+		enc.ChainID = (*hexutil.Big)(tx.ChainID)
+		if tx.Transactions != nil {
```

Comment:
could also do this if `tx.Transactions == nil`

Reply:
mhh, I was taking this from the `copy()` code. I think technically you are right and the tx-data will always be initialised by a "copy()" call so this is never nil. But different than the other type-casts `(e.g. (*hexutil.Big)(tx.Timestamp)` this does panic (at `len(nil)`) if the field would be uninitialised .

Reply:
oh, right, it shouldn't panic. You could initialize it with length 0 and use `append` in the loop. But I'm also fine with leaving it as is. I was mostly thinking it would be nice to always include the field, even if it's empty/not initialized, but it's not important.